### PR TITLE
String buffer

### PR DESCRIPTION
**Pull request by hwhelchel**
StringBuffers are used in Java to limit the number of string objects created in memory when concatenating strings.

In Ruby we can use the String#concat or String#<< methods to update strings in place so there is not a need for StringBuffers as far as I can see. It also looks like the StringIO class might serve a similar purpose.

This will add a Ruby implementation of a Java StringBuffer to emulate the behavior.

### CODE REVIEW

File: spec/data-structures/string-buffer/string_buffer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require_relative '../../spec_helper'
+describe StringBuffer do
+
+  let(:my_buffer) { StringBuffer.new }
+  let(:my_word) { 'test' }
```
Comment (edshadi) [inline spec/data-structures/string-buffer/string_buffer_spec.rb]: generally avoid naming things my_*, I think it's better if you named them buffer and word

File: spec/data-structures/string-buffer/string_buffer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  describe '#initialize' do
+    it 'makes a StringBuffer' do
+      expect(my_buffer).to be_an_instance_of StringBuffer
```
Comment (edshadi) [inline spec/data-structures/string-buffer/string_buffer_spec.rb]: This is testing ruby, we trust that when I initialize a class the result is then an instance of that class. I would remove that test.

File: spec/data-structures/string-buffer/string_buffer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        my_buffer.append my_word
+      }.to change{ buffer.count }.by 1
+    end
+  end
+
+  describe '#to_s' do
+    context 'with a populated buffer' do
+      before(:each) { my_buffer.append my_word }
+      it 'returns a string' do
+        expect(my_buffer.to_s).to be_an_instance_of String
```
Comment (edshadi) [inline spec/data-structures/string-buffer/string_buffer_spec.rb]: to_s always defaults to a string, you can't even force it to return anything else, try it.

**Comment by edshadi** [comment]
looks great, I added a couple of minor comments. Let me know what you think.

**Comment by hwhelchel** [comment]
Hey @edshadi thanks for the comments. I updated my variable names in my tests and removed the tests which tested the data type.

In regards to #to_s, I overrode the default #to_s implementation to return the buffer array of strings as a single concatenated string. I left in this implementation but removed the testing of its data type.

**Comment by edshadi** [comment]
nice dude!

**Pull request merged**